OnPod/Node/Common/PeripheralInterfaces/include/I2CInterface.h

```diff
@@ -0,0 +1,53 @@
+#ifndef H_I2CINTERFACE
+#define H_I2CINTERFACE
+
+#include "main.h"
+
+class I2CInterface
+{
+    private:
+
+    // Pointer to I2C Handle Type Def structure contains the config info for the specified I2C.
+    I2C_HandleTypeDef *m_hi2c;
+    uint8_t m_dataRecieved[8]; 
+    uint8_t m_dataBufferLength = 8;
+    uint32_t m_timeout; 
+
+    public:
+
+    // Constructor
+    I2CInterface(I2C_HandleTypeDef *hi2c);
+    
+    // Returns a pointer to the data recieved buffer
+    uint8_t* getRecievedData() {  return m_dataRecieved; }
+
+    // Get state, and error methods
+    HAL_I2C_StateTypeDef getState() { return HAL_I2C_GetState(m_hi2c); }
+    uint32_t getError() { return HAL_I2C_GetError(m_hi2c); }
+
+    // Getter and setter for I2C addressing mode
+    void setAddressingMode(uint32_t mode) { (*m_hi2c).Init.AddressingMode = mode; }
```

Comment:
@stuckatmarine how do you feel about defining one-liners in the header? Maybe only getters and attribute setters?

Reply:
This should be fine, you can keep it